### ISSUE TITLE
di-order is case sensitive: it enforces upperFirst services to be placed before lowerFirst

### ISSUE THREAD

**Issue by jrencz** [Gillespie59/eslint-plugin-angular#375]
Consider the following snippets:

```js
// ordered alphabetically, no error
this.$get = function (
  $q,
  $rootScope,
  analytics,
  authentication
) {}
```

```js
// Authentication instead of authentication: error
// Injected values should be sorted alphabetically  angular/di-order
this.$get = function (
  $q,
  $rootScope,
  analytics,
  Authentication
) {}
```

```js
// Authentication moved up, above analytics: no error but they are no longer in an alphabetical order
this.$get = function (
  $q,
  $rootScope,
  Authentication,
  analytics
) {}
```

It should be fixed or configurable.
It should definitely be documented.

**Comment by laurent-le-graverend** [comment]
How about an option to have **Natural sort** order instead of "Pure" alphabetical order?

**Comment by laurent-le-graverend** [comment]
@Gillespie59, possible to escalate this issue?

**Comment by Gillespie59** [comment]
I have just started working on this rule

**Comment by Gillespie59** [comment]
I have juste pushed an update for this rule. I have added a second option with these possible values : 'case_sensitive' (the default value, in order to be retro compatible), and 'case_insensitive'. 

Any feedbacks are welcomed.

**Comment by laurent-le-graverend** [comment]
Great , your commit looks good for me @Gillespie59 !

**Comment by Gillespie59** [comment]
perfect. Will publish the 1.3.0 release in a couple of hours.

**Comment by Gillespie59** [comment]
1.3.0 published
